Lib/ast.py

```diff
@@ -666,11 +667,16 @@ def buffer(self):
         return value
 
     @contextmanager
-    def block(self):
+    def block(self, node=None, *, append_type_comment = False):
```

Comment:
This function is doing too much.

 Its responsibility is managing the source for blocks, it should not need to understand or deal with type comments

Reply:
There is no way we can intervene between colon and the new line after that.

Reply:
Yes, the function can receive an optional, already calculated type comment and then it blindly appends it. This way the logic becomes uncoupled and the "set up block" function does not need to graduate into how to calculate type comments